tests/integration/io-threads.tcl

```diff
@@ -0,0 +1,90 @@
+proc get_io_threads_active_num {client} {
+    set server_info [$client info server]
+    regexp {io_threads_active_num:([0-9]+)} $server_info ->\
+        io_threads_active_num
+    return $io_threads_active_num
+}
+
+proc start_benchmark {thread_num client_num} {
+    set bench_pid [
+        exec src/valkey-benchmark -s [srv unixsocket] -t set \
+        -r 5000000 -n 10000000 --threads $thread_num -c $client_num \
+        > /dev/null &
+    ]
+    return $bench_pid
+}
+
+tags {"io-threads external:skip"} {
+    test "Disable io-threads" {
+        start_server {overrides {"io-threads" {1} "io-threads-do-reads" {no}}} {
+            set master_pid [srv 0 pid]
+            set pstree_out [exec ps -T -p $master_pid]
+            assert_no_match "*io_thd_*" $pstree_out
+            assert_match "*io_threads_active:0*" [r info server]
+            assert_match "*io_threads_maximum_num:1*" [r info server]
+            assert_match "*io_threads_active_num:1*" [r info server]
+
+            assert_equal {OK} [r set k v]
+            assert_equal {1} [r DBSIZE]
+        }
+    }
+
+    test "Enable io-threads" {
+        start_server {overrides {"io-threads" {4} "io-threads-do-reads" {yes}}} {
+            set master_pid [srv 0 pid]
+            set pstree_out [exec ps -T -p $master_pid]
+            assert_match "*io_thd_1*" $pstree_out
+            assert_match "*io_thd_2*" $pstree_out
+            assert_match "*io_thd_3*" $pstree_out
+            set server_info [r info server]
+            regexp {io_threads_active:([0-9]+)} $server_info -> \
+                io_threads_active
+            regexp {io_threads_maximum_num:([0-9]+)} $server_info -> \
+                io_threads_maximum_num
+            regexp {io_threads_active_num:([0-9]+)} $server_info -> \
+                io_threads_active_num
+            assert {$io_threads_active == 0}
+            assert {$io_threads_maximum_num == 4}
+            assert {$io_threads_active_num == 1}
+
+            assert_equal {OK} [r set k v]
+            assert_equal {1} [r DBSIZE]
+        }
+    }
+
+    test "Test scale up and scale back down io_threads_active_num." {
+        start_server {overrides {"io-threads" {4} "io-threads-do-reads" {yes}}} {
+            set client [valkey [srv host] [srv port] 0 $::tls]
+            # When the server has just started, no requests from client.
+            # The active threads number should be 1.
+            assert_equal {1} [get_io_threads_active_num $client]
+
+            # Scale up io_threads_active_num by sending large number requests to server.
+            set bench_pid [start_benchmark 10 50]
+            wait_for_condition 1000 500 {
```

Comment:
This is quite a loop but judging by your test runs, this test doesn't seem to increase the test execution time.

Another concern of mine is the potential flakiness of this test since it has non-deterministic dependency in nature. 

That said, I am fine with keeping this test for now. Push comes shove, we can remove it since we will get coverage from #397 as well.

Reply:
> this test doesn't seem to increase the test execution time

Depends on the write speed,  the condition of meet total count of 10000 should be very fast.

>Another concern of mine is the potential flakiness of this test since it has non-deterministic dependency in nature.
>
>That said, I am fine with keeping this test for now. Push comes shove, we can remove it since we will get coverage from https://github.com/valkey-io/valkey/issues/397 as well.

Got it.